### PR TITLE
Master upgrade procedure

### PR DESCRIPTION
**Pull request by muhamadazmy**
This PR concerns only with the following:
- Specifications of the upgrade/fresh boot procedure
- Data migration of the daemons

Another PR will be created for `bootstraping` the node

Part Of #170

### CODE REVIEW

File: modules/identity/keys.go [hunk 1]
```diff
@@ -37,17 +44,36 @@ func GenerateKeyPair() (k KeyPair, err error) {
...
 // KeyPair using the seed
-func LoadSeed(path string) (k KeyPair, err error) {
-	seed, err := ioutil.ReadFile(path)
+func LoadKeyPair(path string) (k KeyPair, err error) {
+	seed, err := LoadSeed(path)
 	if err != nil {
 		return k, err
 	}
-
+	// version update can go here.
```
Comment (zaibon) [inline modules/identity/keys.go]: How do we know the version of the seed we just loaded at this point ? I don't see how we can decide to do an ugprade or not here.

**Review by zaibon — COMMENTED** [review]
LGTM, just had one question

File: modules/identity/keys.go (hunk 1 above)
Comment (muhamadazmy) [reply modules/identity/keys.go]: Ah, I moved the code but didn't move the comment :/

File: modules/versioned/data_test.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+	dec := json.NewDecoder(reader)
+
+	if reader.Version().Compare(latest) <= 0 {
+		//data version is older than or equal latest
+		var data string
+		if err := dec.Decode(&data); err != nil {
+			panic(err)
+		}
+		fmt.Println("data is:", "my data goes here")
+	}
```
Comment (zaibon) [inline modules/versioned/data_test.go:178]: can we add an example how we would migrate the data if the version is not the one expected ?

Comment (muhamadazmy) [reply modules/versioned/data_test.go:178]: sure

**Review by zaibon — APPROVED** [review]

**Comment by maxux** [comment]
This breaks `tools/tfuser` build.

**Pull request merged**